src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVTYPE.java

```diff
@@ -0,0 +1,169 @@
+
+/**
+ * Created by shuang on 2/7/17.
+ */
+abstract class SVTYPE {
+
+    enum RECOGNIZED_SVTYPES {
+        INV, DEL, INS, DUP;
+    }
+
+    abstract String getVariantId();
+
+    abstract Allele getAltAllele();
+
+    abstract int getSVLength();
+
+
+    static final class INV extends SVTYPE {
+        private final String variantId;
+        private final Allele altAllele;
+        private final int svLen;
+
+        String getVariantId() {
+            return variantId;
+        }
+        Allele getAltAllele() {
+            return altAllele;
+        }
+        int getSVLength() {
+            return svLen;
+        }
+        @Override
+        public String toString() {
+            return RECOGNIZED_SVTYPES.INV.name();
+        }
+
+        INV(final NovelAdjacencyReferenceLocations novelAdjacencyReferenceLocations) {
+            final String contig = novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getContig();
+            final int start = novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getEnd();
+            final int end = novelAdjacencyReferenceLocations.leftJustifiedRightRefLoc.getStart();
+            final NovelAdjacencyReferenceLocations.EndConnectionType endConnectionType = novelAdjacencyReferenceLocations.endConnectionType;
+
+            variantId = (endConnectionType == FIVE_TO_FIVE ? GATKSVVCFHeaderLines.INV55 : GATKSVVCFHeaderLines.INV33) + SVConstants.CallingStepConstants.VARIANT_ID_FIELD_SEPARATOR +
+                    contig + SVConstants.CallingStepConstants.VARIANT_ID_FIELD_SEPARATOR + start + SVConstants.CallingStepConstants.VARIANT_ID_FIELD_SEPARATOR + end;
+
+            altAllele = Allele.create(SVConstants.CallingStepConstants.VCF_ALT_ALLELE_STRING_INV);
+
+            svLen = end - start;
+        }
+    }
+
+    static final class DEL extends SVTYPE {
+        private final String variantId;
+        private final Allele altAllele;
+        private final int svLen;
+
+        String getVariantId() {
+            return variantId;
+        }
+        Allele getAltAllele() {
+            return altAllele;
+        }
+        int getSVLength() {
+            return svLen;
+        }
+        @Override
+        public String toString() {
+            return RECOGNIZED_SVTYPES.DEL.name();
+        }
+
+        DEL(final NovelAdjacencyReferenceLocations novelAdjacencyReferenceLocations) {
+            final String contig = novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getContig();
+            final int start = novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getEnd();
+            final int end = novelAdjacencyReferenceLocations.leftJustifiedRightRefLoc.getStart();
+
+            final String startString;
+            if (novelAdjacencyReferenceLocations.complication.dupSeqForwardStrandRep.isEmpty()) {
+                startString = toString();
+            } else {
+                startString = "DUP" +
```

Comment:
As discussed in person, let's start the id with "DEL" or "INS" and then have the repeat expansion or contraction information afterwards.